### PR TITLE
PR: Breaking Change - IntuneDeviceEntollmentPlatformRestriction

### LINKED ISSUE #2347: IntuneDeviceEnrollmentPlatformRestriction not applying any configuration despite showing as successful
#### Details of the scenario you tried and the problem that is occurring
When running a .ps1 file in a pipeline with the IntuneDeviceEnrollmentPlatformRestriction resource, all jobs complete successfully with no errors but the changes do not take affect within the tenant it is being ran against. 
I have followed the example found [here](https://microsoft365dsc.com/resources/intune/IntuneDeviceEnrollmentPlatformRestriction/#example-1) and amended Android, iOS and Windows mobile devices to be blocked. Also Windows Personal Device Enrollment Blocked.
These are currently set as allow within the tenant, but I don't see anything change once the pipeline has been 'successfully' ran.
Other configurations have successfully ran against the tenant, so I know that the pipeline can successfully connect. 
I have taken a manual approach and successfully applied the configuration through both MS Graph and PowerShell using Update-MgDeviceManagementDeviceEnrollmentConfiguration

#### Verbose logs showing the problem
N/A
#### Suggested solution to the issue
Is it possible that [this](https://microsoft365dsc.com/resources/intune/IntuneDeviceEnrollmentPlatformRestriction/#example-1) way of doing things is no longer valid and there is now a new way to deploy the code?
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
Microsoft365 DSC - IntuneDeviceEnrollmentPlatformRestriction

# insert configuration here

Configuration ConfigureMicrosoft365
{
    
    param
    ()

    
    Import-DscResource -ModuleName Microsoft365DSC

    node localhost
    {   
        IntuneDeviceEnrollmentPlatformRestriction Default_Device_Enrollment_Platform_Restriction
        {
            AndroidOSMaximumVersion                      = "";
            AndroidOSMinimumVersion                      = "";
            AndroidPersonalDeviceEnrollmentBlocked       = $False;
            AndroidPlatformBlocked                       = $True;
            Description                                  = "This is the default Device Type Restriction applied with the lowest priority to all users regardless of group membership.";
            DisplayName                                  = "All users and all devices";
            Ensure                                       = "Present";
            iOSOSMaximumVersion                          = "";
            iOSOSMinimumVersion                          = "";
            iOSPersonalDeviceEnrollmentBlocked           = $False;
            iOSPlatformBlocked                           = $True;
            MacPersonalDeviceEnrollmentBlocked           = $False;
            MacPlatformBlocked                           = $False;
            WindowsMobileOSMaximumVersion                = "";
            MacOSMinimumVersion                          = "";
            MacOSMaximumVersion                          = "";
            WindowsMobileOSMinimumVersion                = "";
            WindowsMobilePersonalDeviceEnrollmentBlocked = $False;
            WindowsMobilePlatformBlocked                 = $True;
            WindowsOSMaximumVersion                      = "";
            WindowsOSMinimumVersion                      = "";
            WindowsPersonalDeviceEnrollmentBlocked       = $True;
            WindowsPlatformBlocked                       = $False;
            ApplicationId                                = $env:applicationID;
            TenantId                                     = $env:tenantID;
            CertificateThumbprint                        = $env:certificateThumbprint;
        }
    }
}
#### Version of the DSC module that was used ('dev' if using current dev branch)
Microsoft365DSC = '1.22.921.1'

Appreciate any help I can get on this, 
Cheers

### PR DESCRIPTION
**Pull request by William-Francillette**
#### This Pull Request (PR) fixes the following issues

Rework of IntuneDeviceEntollmentPlatformRestriction:

- Adding descriptions
- Using CIM embedded instances
- Switch to Beta API
- support for singlePlatformRestriction
- Support for additional managed platforms e.g. androidForWork
- Support for Assignments 
- Fixes #2347

### CODE REVIEW

**Comment by ykuijs** [comment]
@William-Francillette Could you please update this PR with Dev, so we can merge it for the Breaking Change release?

**Review by NikCharlebois — APPROVED** [review]

**Comment by William-Francillette** [comment]
Should be fixed

**Comment by ykuijs** [comment]
The example is not correct and is preventing the quality checks to complete successfully. @William-Francillette Could you have another check?

**Comment by William-Francillette** [comment]
@ykuijs Example updated

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceEnrollmentPlatformRestriction/MSFT_IntuneDeviceEnrollmentPlatformRestriction.schema.mof [hunk 1]
```diff
@@ -1,28 +1,40 @@
...
+ [Write, Description("Max OS version supported.")] String OsMaximumVersion;
+ [Write, Description("Collection of blocked Manufacturers.")] String BlockedManufacturers[];
+ [Write, Description("Collection of blocked Skus.")] String BlockedSkus[];
+};
+
 [ClassVersion("1.0.0.0"), FriendlyName("IntuneDeviceEnrollmentPlatformRestriction")]
 class MSFT_IntuneDeviceEnrollmentPlatformRestriction : OMI_BaseResource
 {
-    [Key, Description("Display name of the device enrollment platform restriction.")] String DisplayName;
+    [Write, Description("Identity of the device enrollment platform restriction.")] String Identity;
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceEnrollmentPlatformRestriction/MSFT_IntuneDeviceEnrollmentPlatformRestriction.schema.mof:25]: Please make this property Key

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceEnrollmentPlatformRestriction/MSFT_IntuneDeviceEnrollmentPlatformRestriction.schema.mof [hunk 2]
```diff
@@ -1,28 +1,40 @@
...
+    [Write, Description("Display name of the device enrollment platform restriction.")] String DisplayName;
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceEnrollmentPlatformRestriction/MSFT_IntuneDeviceEnrollmentPlatformRestriction.schema.mof:26]: And this property Required

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceEnrollmentPlatformRestriction/MSFT_IntuneDeviceEnrollmentPlatformRestriction.psm1 [hunk 3]
```diff
@@ -4,93 +4,58 @@ function Get-TargetResource
...
-
-        [Parameter()]
-        [System.Boolean]
-        $AndroidPersonalDeviceEnrollmentBlocked,
-
-        [Parameter()]
-        [System.String]
-        $AndroidOSMinimumVersion,
-
         [Parameter()]
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceEnrollmentPlatformRestriction/MSFT_IntuneDeviceEnrollmentPlatformRestriction.psm1:7]: Parameter should be mandatory

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceEnrollmentPlatformRestriction/MSFT_IntuneDeviceEnrollmentPlatformRestriction.psm1 [hunk 4]
```diff
@@ -4,93 +4,58 @@ function Get-TargetResource
...
-        [Parameter()]
-        [System.Boolean]
-        $iOSPlatformBlocked,
-
-        [Parameter()]
-        [System.Boolean]
-        $iOSPersonalDeviceEnrollmentBlocked,
+        $Identity,
 
         [Parameter()]
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceEnrollmentPlatformRestriction/MSFT_IntuneDeviceEnrollmentPlatformRestriction.psm1:11]: Parameter should be mandatory

**Review by ykuijs — CHANGES_REQUESTED** [review]
Please also add an entry to the changelog

**Comment by William-Francillette** [comment]
@ykuijs should be good now

**Review by ykuijs — APPROVED** [review]
LGTM

**Comment by William-Francillette** [comment]
@ykuijs just fixed unit test
image

**Review by ykuijs — APPROVED** [review]
LGTM

**Pull request merged**